src/Microsoft.Extensions.SecretManager.Tools/Internal/MsBuildProjectFinder.cs

```diff
@@ -33,15 +33,18 @@ public string FindMsBuildProject(string project)
 
             if (Directory.Exists(projectPath))
             {
-                var projects = Directory.EnumerateFileSystemEntries(projectPath, "*.*proj", SearchOption.TopDirectoryOnly).ToList();
+                var projects = Directory.EnumerateFileSystemEntries(projectPath, "*.*proj", SearchOption.TopDirectoryOnly)
+                    .Where(f => !".xproj".Equals(Path.GetExtension(f), StringComparison.OrdinalIgnoreCase))
```

Comment:
Instead of filtering out .xproj, why not just allow the extensions that you know about?